### ISSUE TITLE
ngModel on input of type "range" casts the variable to string

### ISSUE THREAD

**Issue by lazarljubenovic** [angular/angular#11325]
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Binding to an `input` of `type="range"` casts the variable to a string.

**Expected/desired behavior**
It should be a `number`.

**Reproduction of the problem**
I don't think there's a need for a Plunker, though I can make one if this is not enough. Here's a gif instead with relevant code.

![](http://i.imgur.com/vzdsq2i.gif)

```
public typeof(whatever: any): string { return typeof whatever; }

a: number = 100;
b: number = 0.5;

<input type="number" [(ngModel)]="a"> {{a}} {{typeof(a)}}
<input type="range" [(ngModel)]="b"> {{b}} {{typeof(b)}}
```

**What is the expected behavior?**
It should stay a number after going "through" `ngModel`.

**What is the motivation / use case for changing the behavior?**
A number is a number.

I ended up concatenating strings instead of adding something I expected to be a number.

**Please tell us about your environment:**
- **Angular version:** 2.0.0-rc.6
- **Browser:** Chrome 53.0.2785.92, Firefox 48.0 on Ubuntu 16.04
- **Language:** TypeScript 1.8.10

**Comment by AlmogShaul** [comment]
This is the default behavior of the basic range element 
It is not related to Angular, I think

**Comment by skreborn** [comment]
As per @AlmogShaul's comment, this is working as intended, and has nothing to do with Angular. Read up on the [`HTMLInputElement` on MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement), or check out [this fiddle](https://jsfiddle.net/jtuoxtms/).

> **[value]** *string*: Returns / Sets the current value of the control.

**Comment by AlmogShaul** [comment]
Can I open a pr for it?

**Comment by vicb** [comment]
> Can I open a pr for it?

PR are always welcome

**Comment by AlmogShaul** [comment]
#11792

**Comment by CITguy** [comment]
Is this something that will be backported to 1.x? I just ran into this problem.

**Comment by DzmitryShylovich** [comment]
is this still an issue after https://github.com/angular/angular/pull/11792?

**Comment by CITguy** [comment]
@DzmitryShylovich: #11792 applied a fix to Angular 2. Can we use this issue to track back-porting the fix to 1.x or should I create another issue to track it?

**Comment by AlmogShaul** [comment]
@DzmitryShylovich this issue can be closed  #11792 fixed it.
